src/core/olimageryprovider.js

```diff
@@ -208,12 +208,9 @@ olcs.core.OLImageryProvider.prototype.requestImage = function(x, y, level) {
   var tileUrlFunction = this.source_.getTileUrlFunction();
   if (!goog.isNull(tileUrlFunction) && !goog.isNull(this.projection_)) {
     // perform mapping of Cesium tile coordinates to ol3 tile coordinates
-    var z_ = (this.tilingScheme_ instanceof Cesium.GeographicTilingScheme) ?
-             (level + 1) : level;
-    var y_ = (this.source_.getTileGrid() instanceof ol.tilegrid.XYZ) ?
-             y : (y - (1 << level));
-    y_ = -y_ - 1; // opposite indexing
-    var url = tileUrlFunction([z_, x, y_], 1, this.projection_);
+    var transform = this.source_.getTileGrid().createTileCoordTransform();
```

Comment:
I implemented your suggestion but had to cast the result from `this.source_.getTileGrid().createTileCoordTransform();`. I guess calling a member function is somehow incorrect/unsupported and we should access it through the prototype. I will add it to Ol3.